yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtilSuite.scala

```diff
@@ -245,4 +247,28 @@ class YarnSparkHadoopUtilSuite extends SparkFunSuite with Matchers with Logging
       System.clearProperty("SPARK_YARN_MODE")
     }
   }
+
+  test("Obtain tokens For HiveMetastore") {
+    val hadoopConf = new Configuration()
+    hadoopConf.set("hive.metastore.kerberos.principal", "bob")
+    // thrift picks up on port 0 and bails out, without trying to talk to endpoint
+    hadoopConf.set("hive.metastore.uris", "http://localhost:0")
+    val util = new YarnSparkHadoopUtil
+    val e = intercept[InvocationTargetException] {
+      val token = util.obtainTokenForHiveMetastoreInner(hadoopConf, "alice")
+      fail(s"Expected an exception, got the token $token")
```

Comment:
As with the case below, I fail to see how the token can help in figuring out what was wrong. If a token is returned, you kinda know what it will be already. You're testing the fact that a token should not be returned at all, and an exception thrown instead. If the exception is not thrown (or the wrong one), that's the problem, and the returned token won't help in figuring out why.